cloud/cloudstack/cs_loadbalancer_rule_member.py

```diff
@@ -0,0 +1,252 @@
+      - Domain the rule is related to.
+    required: false
+    default: null
+  account:
+    description:
+      - Account the rule is related to.
+    required: false
+    default: null
+  zone:
+    description:
+      - Name of the zone in which the rule shoud be located.
+      - If not set, default zone is used.
+    required: false
+    default: null
+extends_documentation_fragment: cloudstack
+'''
+
+EXAMPLES = '''
+# Add VMs to Load Balancer
+- local_action:
+    module: cs_loadbalancer_rule_member
+    name: balance_http
+    vms:
+      - VM1
+      - VM2
+
+# Remove VMs from Load Balancer
+- local_action:
+    module: cs_loadbalancer_rule_member
+    name: balance_http
+    vms:
+      - VM3
+      - VM4
+    state: absent
+
+# Rolling upgrade of hosts
+- hosts: webservers
+  serial: 1
+  pre_tasks:
+    - name: Remove from load balancer
+      local_action:
+      module: cs_loadbalancer_rule_member
+      name: balance_http
+      vms:
+        - "{{ ansible_hostname }}"
+      state: absent
+  tasks:
+    # Perform update
+  post_tasks:
+    - name: Add to loadbalancer
+      local_action:
+      module: cs_loadbalancer_rule_member
+      name: balance_http
+      vms:
+        - "{{ ansible_hostname }}"
+      state: present
+'''
+
+RETURN = '''
+---
+id:
+  description: UUID of the rule.
+  returned: success
+  type: string
+  sample: a6f7a5fc-43f8-11e5-a151-feff819cdc9f
+zone:
+  description: Name of zone the rule is related to.
+  returned: success
+  type: string
+  sample: ch-gva-2
+project:
+  description: Name of project the rule is related to.
+  returned: success
+  type: string
+  sample: Production
+account:
+  description: Account the rule is related to.
+  returned: success
+  type: string
+  sample: example account
+domain:
+  description: Domain the rule is related to.
+  returned: success
+  type: string
+  sample: example domain
+'''
+
+
+try:
+    from cs import CloudStack, CloudStackException, read_config
+    has_lib_cs = True
+except ImportError:
+    has_lib_cs = False
+
+# import cloudstack common
+from ansible.module_utils.cloudstack import *
+
+
+class AnsibleCloudStackLBRuleMember(AnsibleCloudStack):
+
+    def get_rule(self, **kwargs):
+        rules = self.cs.listLoadBalancerRules(**kwargs)
+        if rules:
+            return rules['loadbalancerrule'][0]
+
+    def _get_common_args(self):
+        return {
+            'account': self.get_account(key='name'),
+            'domainid': self.get_domain(key='id'),
+            'zoneid': self.get_zone(key='id'),
+        }
+
+    def _change_members(self, operation):
+        if operation not in ['add', 'remove']:
+            self.module.fail_json(msg="Bad operation: %s" % operation)
+        args = self._get_common_args()
+        rule = self.get_rule(name=self.module.params.get('name'), **args)
+        if not rule:
+            self.module.fail_json(msg="Unknown rule: %s" % self.module.params.get('name'))
+        if not self.module.check_mode:
```

Comment:
imho this condition is not on the right place. check mode should be right before the execution of removeFromLoadBalancerRule() or assignToLoadBalancerRule(). otherwise --check would always return `changed: false`